server/src/test/java/org/elasticsearch/action/admin/indices/alias/GetAliasesResponseTests.java

```diff
@@ -0,0 +1,245 @@
+        switch (randomInt(2)) {
+        case 0:
+            return new GetAliasesResponse(mutateAliases(response.getAliases()), response.status(), response.errorMessage());
+        case 1:
+            return new GetAliasesResponse(response.getAliases(),
+                    randomValueOtherThan(response.status(), () -> randomFrom(RestStatus.values())), response.errorMessage());
+        case 2:
+            if (response.status() == RestStatus.OK) {
+                return new GetAliasesResponse(response.getAliases(),
+                        randomValueOtherThan(response.status(), () -> randomFrom(RestStatus.values())), randomAlphaOfLengthBetween(5, 100));
+            }
+            return new GetAliasesResponse(response.getAliases(), response.status(), randomAlphaOfLengthBetween(5, 100));
+        default:
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    private static ImmutableOpenMap<String, List<AliasMetaData>> mutateAliases(ImmutableOpenMap<String, List<AliasMetaData>> aliases) {
+        if (aliases.isEmpty()) {
+            return createIndicesAliasesMap(1, 3).build();
+        }
+
+        if (randomBoolean()) {
+            ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder(aliases);
+            ImmutableOpenMap<String, List<AliasMetaData>> list = createIndicesAliasesMap(1, 2).build();
+            list.forEach(e -> builder.put(e.key, e.value));
+            return builder.build();
+        }
+
+        Set<String> indices = new HashSet<>();
+        Iterator<String> keys = aliases.keysIt();
+        while (keys.hasNext()) {
+            indices.add(keys.next());
+        }
+
+        List<String> indicesToBeModified = randomSubsetOf(randomIntBetween(1, indices.size()), indices);
+        ImmutableOpenMap.Builder<String, List<AliasMetaData>> builder = ImmutableOpenMap.builder();
+
+        for (String index : indices) {
+            List<AliasMetaData> list = new ArrayList<>(aliases.get(index));
+            if (indicesToBeModified.contains(index)) {
+                if (randomBoolean() || list.isEmpty()) {
+                    list.add(createAliasMetaData());
+                } else {
+                    int aliasIndex = randomInt(list.size() - 1);
+                    AliasMetaData aliasMetaData = list.get(aliasIndex);
+                    list.add(aliasIndex, mutateAliasMetaData(aliasMetaData));
+                }
+            }
+            builder.put(index, list);
+        }
+        return builder.build();
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return p -> p.equals("") // do not add elements at the top-level as any element at this level is parsed as a new index
+                || p.endsWith(".aliases") // do not add new alias
+                || p.contains(".aliases."); // do not insert random fields in AlilasMetaData.fromContent
```

Comment:
I have some issues parsing `AliasMetaData` once random data is inserted into it. I will investigate this tomorrow.

Reply:
sure, let me know if I can do anything to help!

Reply:
Hey @javanna! The issue turned out to be in the `AliasMetaData#fromXContent`. I just pushed the change, so that the `GetAliasesResponseTests` pass. But now thinking about it, maybe it would be better to create a mini PR only for this change ? Not to hide it in this messy one. What do you think ?

Reply:
great catch thank you! It could be pushed separately, but I don't mind if it's part of this PR given that the new test uncovered this.